pac4j-oauth/src/main/java/org/pac4j/oauth/client/BaseOAuth20StateClient.java

```diff
@@ -33,18 +30,13 @@ protected String getStateParameter(final WebContext context) {
     }
 
     @Override
-    protected String retrieveAuthorizationUrl(final WebContext context) throws RequiresHttpAction {
+    protected void internalInit(WebContext context) {
```

Comment:
The state may be a random string, so we cannot have only one instance of the service to generate the authorization url. We need to re-create a new service every time (for each new state parameter).

I don't think this will work.